app/helpers/automated_tests_helper.rb

```diff
@@ -136,96 +136,59 @@ def process_test_form(assignment, params)
     # Retrieve all test scripts
     testscripts = params[:test_scripts_attributes]
 
-    # First check for duplicate script names in test scripts
-    unless testscripts.nil?
-      testscripts.values.each do |tfile|
-        if tfile['script_name'].respond_to?(:original_filename)
-          fname = tfile['script_name'].original_filename
-          # If this is a duplicate script name, raise error and return
-          if !file_name_array.include?(fname)
-            file_name_array << fname
-          else
-            raise I18n.t('automated_tests.duplicate_filename') + fname
-          end
-        end
-      end
-    end
-
     # Retrieve all test support files
     testsupporters = params[:test_support_files_attributes]
 
-    # Now check for duplicate file names in test support files
-    unless testsupporters.nil?
-      testsupporters.values.each do |tfile|
-        if tfile['file_name'].respond_to?(:original_filename)
-          fname = tfile['file_name'].original_filename
-          # If this is a duplicate file name, raise error and return
-          if filename_array.include?(fname)
-            raise I18n.t('automated_tests.duplicate_filename') + fname
-          else
-            filename_array << fname
-          end
-        end
-      end
-    end
-
-    # Filter out script files that need to be created and updated
+    # Create/Update test scripts
     unless testscripts.nil?
-      testscripts.each_key do |key|
+      testscripts.each do |file_num, file|
+        updated_script_files[file_num] = {}
 
-        tfile = testscripts[key]
+        # Empty file submission, skip
+        next if testscripts[file_num][:script_name].nil?
```

Comment:
Sorry did you mean just add it to the `assignment_params`?

```
def assignment_params
    params.require(:assignment)
        .permit(:enable_test,
                :assignment_id,
                :tokens_per_day,
                :unlimited_tokens,
                test_files_attributes:
                    [:id, :filename, :filetype, :is_private, :_destroy],
                test_scripts_attributes:
                    [:assignment_id, :seq_num, :script_name, :description,
                     :new_script,
                     :max_marks, :run_on_submission, :run_on_request,
                     :halts_testing, :display_description, :display_run_status,
                     :display_marks_earned, :display_input,
                     :display_expected_output, :display_actual_output])
  end
```

Tried that but it's still not being passed through params? Or do you mean add `new_script` as another argument to the `process_test_form` function itself?